core/xml.js

```diff
@@ -471,6 +478,24 @@ Blockly.Xml.domToBlock = function(xmlBlock, workspace) {
 };
 
 /**
+ * Decode an XML list of variables and add the variables to the workspace.
+ * @param {!Element} xmlVariables List of XML variable elements.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ */
+Blockly.Xml.domToVariables = function(xmlVariables, workspace) {
+  for (var i = 0, xmlChild; xmlChild = xmlVariables.children[i]; i++) {
+    var type = xmlChild.getAttribute('type');
+    var id = xmlChild.getAttribute('id');
+    var name = xmlChild.textContent;
+
+    if (!type && type != '') {
```

Comment:
Awesome. I was wondering if there was a  more explicit way to do this. Thank you!

Done.